Add readme with guide for package authors

ericphanson opened the pull request:
I think this might help provide linkable explanations when issues arise during registration. ~~I set it up as a standard Documenter.jl setup, and I think that means a `DOCUMENTER_KEY` environmental variable will need to be added to the travis config for it to work (https://juliadocs.github.io/Documenter.jl/stable/man/hosting/).~~

edit: Now this is just a README.md file

asinghvi17 commented on docs/src/guide.md:
```diff
@@ -0,0 +1,58 @@
+
+The process usually proceeds as follows:
+
+Either
```
```suggestion
Either:

asinghvi17 commented on docs/src/guide.md:
```diff
@@ -0,0 +1,58 @@
+
+After either option, a pull request is made against the [JuliaRegistries/General](https://github.com/JuliaRegistries/General/pulls) GitHub repository.
+
+Often, that is all one needs to do. New packages are subject to a 3-day waiting period for community comment, while new versions of existing packages may be merged right away. Once the pull request is merged, the package or new version of the package is available in the General registry. If TagBog is installed, then it will tag a release once the pull request is merged. This is very helpful to en ...
```
```suggestion
Often, that is all one needs to do. New packages are subject to a 3-day waiting period for community comment, while new versions of existing packages may be merged right away. Once the pull request is merged, the package or new version of the package is available in the General registry. If [TagBot](https://github.com/JuliaRegistries/TagBot) is installed, then it will tag a release once the pull request is merged. This is very helpful to ensure that tagged releases correspond exactly to versions in the registry.

asinghvi17 commented on docs/src/guide.md:
```diff
@@ -0,0 +1,58 @@
+
+## Compatability ("needs compat")
+
+Every Julia package has at least one (implicit) dependency: the Julia language itself. It is recommended that for every dependency, each package declares what versions of the dependency it is compatible with. This is done in the package's `Project.toml` file, and is documented [here](https://julialang.github.io/Pkg.jl/v1/compatibility/) in the `Pkg` standard library's doucmentation.
```
```suggestion
Every Julia package has at least one (implicit) dependency: the Julia language itself. It is recommended that for every dependency, each package declares what versions of the dependency it is compatible with. This is done in the package's `Project.toml` file, and is documented [here](https://julialang.github.io/Pkg.jl/v1/compatibility/) in the Julia package manager's documentation.
```

Might not be necessary, I felt it made it clearer though.

asinghvi17 commented on docs/src/guide.md:
```diff
@@ -0,0 +1,58 @@
+
+## Julia Package naming conventions
+
+The guidelines for naming packages are listed [here](https://julialang.github.io/Pkg.jl/v1/creating-packages/#Package-naming-guidelines-1) in the `Pkg` standard library's documentation. If a new package's name does not adhere to those guidelines or could otherwise cause confusion, maintainers of the General registry or the Julia community might comment on the pull request to ask that the name be  ...
```
```suggestion
The guidelines for naming packages are listed [here](https://julialang.github.io/Pkg.jl/v1/creating-packages/#Package-naming-guidelines-1) in the package manager's documentation. If a new package's name does not adhere to those guidelines or could otherwise cause confusion, maintainers of the General registry or the Julia community might comment on the pull request to ask that the name be changed, or why the usual conventions were not used in this case.

fredrikekre commented:
I would prefer just having this in the README. We don't need a full doc when we don't even have things like docstrings etc, and going into the docs will just be one more click.

ericphanson commented:
Ok, makes sense to me; I made the move.

asinghvi17 commented on README.md:
```diff
@@ -0,0 +1,64 @@
+
+The official registry of general Julia packages. This registry of packages is what connects the `Pkg` standard library (see <https://julialang.github.io/Pkg.jl/v1/getting-started/>) to the large ecosystem of Julia packages, allowing them to be installed and updated easily.
+
+Package authors may find the registry's [Guide for package authors](@ref) helpful.
```
Change this to a link to that section?

asinghvi17 replied:
Ah, you beat me to it :D

ericphanson replied:
I just saw that too-- I just deleted that line, because now the section is just below anyway, so I figure a link isn't needed anymore

ericphanson commented:
I hard-wrapped the lines so that diffs are easier to read if the README gets updated, or this PR gets more review comments.

gaelforget commented:
FYI this readme has already been useful to me. Thanks @ericphanson et al

There are a couple edits / typo fixes I could suggest or contribute if easier. One thought is that maybe a link to this readme could be added at the end of the standard bot message (e.g. https://github.com/gaelforget/IndividualDisplacements.jl/issues/1#issuecomment-531625027).

ericphanson commented:
> FYI this readme has already been useful to me. Thanks @ericphanson et al

Glad to hear it!

> There are a couple edits / typo fixes I could suggest or contribute if easier. 

That would be great-- I think the Github review tool lets you suggest changes and fixes.

>One thought is that maybe a link to this readme could be added at the end of the standard bot message (e.g. [gaelforget/IndividualDisplacements.jl#1 (comment)](https://github.com/gaelforget/IndividualDisplacements.jl/issues/1#issuecomment-531625027)).

I was thinking comments like https://github.com/JuliaRegistries/General/pull/3554#issuecomment-533058432 could be written as "Needs a rebase ([link])" where the link goes to this README at the right section.

gaelforget commented:
> I was thinking comments like [#3554 (comment)](https://github.com/JuliaRegistries/General/pull/3554#issuecomment-533058432) could be written as "Needs a rebase ([link])" where the link goes to this README at the right section.

Makes sense to me too

ericphanson commented:
Ok thanks, I made those changes and fixed a couple other typos I spotted too.

Nosferican commented on README.md:
```diff
@@ -0,0 +1,126 @@
+# General registry
+
+The official registry of general Julia packages. This registry of packages is
```
This seems awkward "of general Julia packages". I like better the one in the registry description,
> Official general Julia package registry where people can
register any package they want without too much debate about
naming and without enforced standards on documentation or
testing. We nevertheless encourage documentation, testing and
some amount of consideration when choosing package names.

ericphanson replied:
I agree it sounds a bit awkward; it is the tagline of the repo and possibly could be changed there as well. I will update the README.

Nosferican commented on README.md:
```diff
@@ -0,0 +1,126 @@
+The official registry of general Julia packages. This registry of packages is
+what connects the `Pkg` standard library (see
+<https://julialang.github.io/Pkg.jl/v1/getting-started/>) to the large ecosystem
+of Julia packages, allowing them to be installed and updated easily.
```
I would maybe describe what a registry is rather than paint it as General being "unique".

ericphanson replied:
I just copied the tagline from the repo, but I'm open to changing the wording here-- any suggestions?

Nosferican commented on README.md:
```diff
@@ -0,0 +1,126 @@
+[JuliaRegistries/General](https://github.com/JuliaRegistries/General/pulls)
+GitHub repository. This is automated by the
+[Registrator.jl](https://github.com/JuliaRegistries/Registrator.jl) GitHub bot
+and the [web interface](https://github.com/JuliaRegistries/Registrator.jl).
```
maybe "or" rather than "and"

Nosferican commented on README.md:
```diff
@@ -0,0 +1,126 @@
+GitHub repository.
+
+Often, that is all one needs to do. New packages are subject to a 3-day waiting
+period for community comment, while new versions of existing packages may be
```
"feedback" rather than "comment"?

ericphanson commented:
I think the registry automerge bot now handles most of the things I tried to help address with this PR, namely by giving guidance to package authors when their registration PR doesn't match all the common requirements. Now these rules / suggestions are codified in the bot, and maybe it's not a good idea to have them also here in the README, since it could get out of date (and whatever the bot does is the source-of-truth since it actually merges PRs).

DilumAluthge commented:
@ericphanson I think it's fine to have the guidelines listed in the General registry README file. If things get out of sync, we can just make pull requests to update the README in General.

DilumAluthge commented:
@ericphanson On a related note, in your README, could you add a link to https://github.com/JuliaRegistries/RegistryCI.jl? And maybe you can add a note saying "To report issues with AutoMerge, please open an issue on the RegistryCI repo (not the General registry repo)"?

ericphanson commented:
@DilumAluthge Ok, good to know it still has some use. I've updated the README to change some of the wording to refer to the automerge bot instead of the maintainers, and added the comment about reporting issues.

oxinabox commented on README.md:
```diff
@@ -0,0 +1,164 @@
+registered, and sticking to the conventions makes it easier for users to
+navigate Julia's many varied packages.
+
+To rename the package from `OldName.jl` to `NewName`.jl,
```
```suggestion
As long as the package is not yet registered; renamiing the package from `OldName.jl` to `NewName`.jl, is reasonably straght-forward:
```

Just so peole are really clear that these instructions  only apply to unregistered packages

ericphanson replied:
Sounds good, I added this clarification

oxinabox commented on README.md:
```diff
@@ -110,7 +110,21 @@ with more than one breaking release of a package, add multiple entries such as
 
-See [the `Pkg`
+This attests that the package is compatible with both versions of `StaticArrays`
+even though, according to [semantic versioning](https://semver.org/), the `0.11`
```
Should we link to the docs explaining julia extension of SemVer instead?

ericphanson replied:
Sure, that makes sense. Do you think this would be the right section? https://julialang.github.io/Pkg.jl/stable/compatibility/#Version-specifier-format-1

oxinabox replied:
We need to link to a newer version, so it includes this section abojut leading zeros
https://julialang.github.io/Pkg.jl/dev/compatibility/#Behavior-of-versions-with-leading-zeros-(0.0.x-and-0.x.y)-1

ericphanson replied:
Ok, done

ericphanson commented:
I think it would be nice if this could get merged sooner than later, assuming it's along the lines of something that would be desired in a readme for this repo. That way it could start to be useful, and any problems or things that need to be clarified more might arise more obviously, and would make for easy PRs for anyone interested in changing something.

DilumAluthge commented:
Bump @fredrikekre 

What’s the status of this? Would be great to get some kind of README in the General registry.

waldyrious commented on README.md:
```diff
@@ -0,0 +1,186 @@
+As long as the package is not yet registered, renaming the package from
+`OldName.jl` to `NewName`.jl is reasonably straightforward:
+
+
```
There's no need for two blank lines here, right?

ericphanson replied:
True! Fixed

ericphanson commented:
I updated the PR to include the full list of automerge requirements, as suggested in https://discourse.julialang.org/t/announcement-automatic-merging-for-the-general-registry/29961/46.

Bump :)

ericphanson commented:
Bump

ericphanson commented:
Bump

DilumAluthge commented:
@ericphanson Can you rebase this on master?

ericphanson commented:
oops, I did it wrong, sorry. Let me try to fix it.

ericphanson commented:
Ok fixed. I wasn't able to save the git history, but I think anyway the feedback is recorded in this PR.

fredrikekre commented:
Merged in 6d922337e872738532800201a75e3bf3a136db7b. I made some edits to not include duplicate information with what is in e.g. Registrators README (it is easy they become out of sync) and also rewrote some parts into a FAQ.

Thanks for your contribution!

ericphanson commented:
Thanks very much @fredrikekre! I appreciate the rewriting as well, it seems more clear and concise this way.
